sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -323,6 +323,20 @@ trait RuntimeReplaceable extends UnaryExpression with Unevaluable {
   // two `RuntimeReplaceable` are considered to be semantically equal if their "child" expressions
   // are semantically equal.
   override lazy val canonicalized: Expression = child.canonicalized
+
+  override def innerChildren: Seq[Expression] = sys.error("RuntimeReplaceable must implement" +
+    " innerChildren with the original parameters")
+
+  protected val sqlStrSeparator: String = ", "
+
+  override def sql: String = RuntimeReplaceable.this.prettyName +
+    prettyChildren.map(_.sql).mkString("(", sqlStrSeparator, ")")
+
+  protected var prettyChildren: Seq[Expression] = innerChildren
```

Comment:
I think its better to user `var` where possible, so how about it like this?
```
  // RuntimeReplaceable
  def newInstance(innerChildren: Seq[Expression]): RuntimeReplaceable

  // e.g., ParseToTimestamp
  override def newInstance(innerChildren: Seq[Expression]): ParseToTimestamp = {
    innerChildren match {
      case Seq(left) => ParseToTimestamp(left, None, child)
      case Seq(left, format) => ParseToTimestamp(left, Some(format), child)
    }
  }
```
Then, update the `toPrettySQL` in `utils.package`;
```
  def toPrettySQL(e: Expression): String = e match {
    case r: RuntimeReplaceable =>
      r.newInstance(r.innerChildren.map(usePrettyExpression)).sql
    case _ => usePrettyExpression(e).sql
  }
```

Reply:
The `newInstance` seems not robust to make a clone with `Seq[Expression]`

I think we can just add  `def prettySQL: String` in RuntimeReplaceable for  `utils.package` to call, and override it wherever needed without creating new instances.

Seems a better and simpler idea that comes up based on your suggestions, WDYT?

Reply:
I feel having the pretty functionality in `RuntimeReplaceable` is a bit intrusive. Dropping the idea above is okay, but I think we need a more general solution for getting pretty strings outside `RuntimeReplaceable`. WDYT? @cloud-fan